storage/src/archive/storage.rs

```diff
@@ -41,19 +31,18 @@ pub struct Archive<T: Translator, K: Array, B: Blob, E: Storage<B> + Metrics> {
     // Oldest allowed section to read from. This is updated when `prune` is called.
     oldest_allowed: Option<u64>,
 
-    // We store the first index of the linked list in the HashMap
-    // to significantly reduce the number of random reads we need to do
-    // on the heap.
     indices: BTreeMap<u64, Location>,
```

Comment:
I suggest something like this:
```rust
// To efficiently serve `get` and `has` requests, we map
// a truncated representation of each key to its corresponding index.
// To avoid iterating over this keys map during pruning, we map said indexes
// to their locations in the journal.
keys: Index<T, u64>,
indices: BTreeMap<u64, Location>,
intervals: RangeInclusiveSet<u64>,
```

Reply:
done!